compiler/optimizations/gpuTransforms.cpp

```diff
@@ -721,6 +749,22 @@ void GpuKernel::populateBody(CForLoop *loop, FnSymbol *outlinedFunction) {
     }
   }
 
+  for_vector(CallExpr, callExpr, callExprsInBody) {
```

Comment:
I think it's fine to leave it inside of the `GpuKernel`, particularly because `GpuKernel` already modifies the AST (it removes the `blockSize` primitive, for example!). I did separate the function into its own method, though.

Reply:
>I think it's fine to leave it inside of the GpuKernel, particularly because GpuKernel already modifies the AST (it removes the blockSize primitive, for example!). I did separate the function into its own method, though.

If I had my preference I would move that outside of `GpuKernel` as well.

I realize you didn't write that code, and it's not the worse thing I've ever seen in compiler code but the fact that merely constructing a `GPUKernel` object can mutate the IR isn't intuitive based on the name alone.

If you keep it in `GpuKernel` can you add some comments to the comments above the declaration to the constructor explaining that these side effects exists?

Reply:
> If you keep it in GpuKernel can you add some comments to the comments above the declaration to the constructor explaining that these side effects exists?

I think I will do this.